mkdocs/docs/protocol/onchain-credential-status.md

```diff
@@ -0,0 +1,170 @@
+# OnChain Credential Status
+
+On-chain identity verification requires the ability to check whether a claim has been revoked using a smart contract. To accomplish this, the claim must include information about the smart contract, such as its address, blockchain, and network. This information can be stored in the `CredentialStatus` structure.
+
+The `CredentialStatus` structure should contain the following fields:
+
+```golang
+type CredentialStatus struct {
+	ID                 string               `json:"id"`
+	Type               CredentialStatusType `json:"type"`
+	RevocationNonce    uint64               `json:"revocationNonce"`
+}
+```
+
+The `id` field is a composite field that contains encoded information in the following format:
+
+`did:[did-method]:[blockchain]:[network]:[id]/credentialStatus?revocationNonce=value&contractAddress={chainID}:{contractAddress}`
+
+`type` - credential type
+
+`revocationNonce` - unique revocation nonce
+
+Example:
+
+```json
+{
+"id": "did:polygonid:polygon:main:2qCU58EJgrEMWhziKqC3qNXJkZPY8XCxDSBM4mqPkM/credentialStatus?revocationNonce=1234&contractAddress=1:0xf3bB959314B5D1e4587e1f597ccc289216608ac5",
+"type": "Iden3OnchainSparseMerkleTreeProof2023",
+"revocationNonce": "1234"
+}
+```
+
+## Process ID field
+
+In the context of `OnChainCredentialStatus`, the `id` field can contain two additional optional parameters: `revocationNonce` and `contractAddress`.
+`revocationNonce` is a credential revocation `nonce`, while `contractAddress` is the address of a smart contract that holds public information of identity such as revocation trees and roots tree, etc. The `contractAddress` field is composed by two parts: `chainID` and `contractAddress`. Use `chainID` to select the correct network.
+
+If `contractAddress` is not set, find the default contract address by parsing DID [extracting the on-chain issuer contract address](https://github.com/iden3/go-iden3-core/blob/014f51e92da5c0c89c95c31e42bfca1652d2ad14/did.go#L345-L354) and getting `chainID` from the DID network. Use the blockchain name, network and contract address from the DID to make on-chain revocation request. If the DID doesn't have a contract address inside and `contractAddress` parameter is empty, this VC document should be considered invalid.
+
+If `revocationNonce` is not set, the `revocationNonce` value from the struct will be used instead.
+
+## Workflow
+
+Example of how to build a `non-revocation` proof with the `Iden3OnchainSparseMerkleTreeProof2023` credential status type:
+
+1. Extract the `credentialStatus` object from the verifiable credential.
```

Comment:
Add a new step on how to get `core.Id` from credentialStatus.Id. Later we should use this id for calling onchan-issuer.